account/manager.go

```diff
@@ -165,10 +166,44 @@ func actionTxLabel(account *Account, action Action, isExpirySpend bool,
 	if err != nil {
 		log.Errorf("Internal error: failed to serialize json "+
 			"from %v: %v", label, err)
-		return fmt.Sprintf(prefixTag, action)
+		return fmt.Sprintf(txLabelPrefixTag, action)
+	}
+
+	return fmt.Sprintf(txLabelPrefixTag, labelJson)
+}
+
+// IsPoolTx returns true if the given transaction is related to
+// pool.
+func IsPoolTx(tx *lnrpc.Transaction) bool {
+	return strings.HasPrefix(tx.Label, txLabelPrefixTag)
+}
+
+// ParseTxLabel parses and returns data fields stored in a given account
+// transaction label.
+func ParseTxLabel(label string) (*AccountTxLabel, error) {
+	prefix := fmt.Sprintf(txLabelPrefixTag, "")
+	label = strings.TrimPrefix(label, prefix)
+
+	var data TxLabel
+	err := json.Unmarshal([]byte(label), &data)
+	if err != nil {
+		return nil, err
 	}
 
-	return fmt.Sprintf(prefixTag, labelJson)
+	return &data.Account, nil
+}
+
+// SelectOutput returns the account specific output detail given a transaction
+// and the account output index.
+func SelectOutput(tx *lnrpc.Transaction, accountOutputIndex uint32) (
+	*lnrpc.OutputDetail, error) {
+
+	for _, output := range tx.OutputDetails {
+		if output.OutputIndex == int64(accountOutputIndex) {
```

Comment:
q: aren't the outputs in order? where `tx.OutputDetails[i].OuptutIndex == i`

Reply:
I wasn't sure. Should I assume that they are?

Reply:
Yes, the outputs are in order, should be able to get rid of the whole function.